File: MultiboxClientCSharpCore/Model/LibraryItems/DirectoryItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;


namespace MultiboxClientCSharpCore.Model.LibraryItems
{
	public class DirectoryItem : GenericLibraryItem
	{
		private IList<ILibraryItem> content;

		public DirectoryItem(long id, string name)
			: base(id, LibraryItemType.Directory, name)
		{
			this.content = new List<ILibraryItem>();
		}

		public void AddItem(ILibraryItem item)
		{
			content.Add(item);
		}

		public IList<ILibraryItem> Items
		{
			get { return new List<ILibraryItem>(content); }
		}
	}
}

File: MultiBoxClientJavaCore/src/sk/hackcraft/multibox/model/libraryitems/DirectoryItem.java
package sk.hackcraft.multibox.model.libraryitems;

import java.util.LinkedList;
import java.util.List;

import sk.hackcraft.multibox.model.GenericLibraryItem;
import sk.hackcraft.multibox.model.LibraryItem;
import sk.hackcraft.multibox.model.LibraryItemType;

public class DirectoryItem extends GenericLibraryItem
{
	private List<LibraryItem> contentIds;
	
	public DirectoryItem(long id, String name)
	{
		super(id, LibraryItemType.DIRECTORY, name);
		
		this.contentIds = new LinkedList<LibraryItem>();
	}
	
	public void addItem(LibraryItem item)
	{
		contentIds.add(item);
	}
	
	public List<LibraryItem> getItems()
	{
		return new LinkedList<LibraryItem>(contentIds);
	}
}
